src/components/Comment/CommentsList.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { useCallback, useContext } from 'react';
+import { Comment } from '../../types/Comment';
+import { DispatchContext } from '../../reducer/store';
+import { commentService } from '../../services/comment.service';
+
+type Props = {
+  comments: Comment[]
+};
+
+export const CommentsList: React.FC<Props> = ({ comments }) => {
+  const dispatch = useContext(DispatchContext);
+
+  const onDeleteButton = useCallback((id: number) => () => {
+    const { removeComment } = commentService(dispatch, comments);
```

Comment:
This line of code initiates a new instance of the comment service on every call to the onDeleteButton function which might not be efficient if many comments are being deleted at once. A new instance should only be created once, preferably in the component's lifecycle method.